lib/managementClient.ts

```diff
@@ -0,0 +1,231 @@
+   * @method getPartitionInformation
+   * @param {Connection} connection - The established amqp connection
+   * @param {(string|number)} partitionId Partition ID for which partition information is required.
+   */
+  async getPartitionInformation(connection: any, partitionId: string | number): Promise<EventHubPartitionRuntimeInformation> {
+    if (!partitionId || (partitionId && typeof partitionId !== "string" && typeof partitionId !== "number")) {
+      throw new Error("'partitionId' is a required parameter and must be of type: 'string' | 'number'.");
+    }
+    const info: any = await this._makeManagementRequest(connection, Constants.partition, partitionId);
+    const partitionInfo: EventHubPartitionRuntimeInformation = {
+      beginningSequenceNumber: info.begin_sequence_number,
+      hubPath: info.name,
+      lastEnqueuedOffset: info.last_enqueued_offset,
+      lastEnqueuedTimeUtc: new Date(info.last_enqueued_time_utc),
+      lastSequenceNumber: info.last_enqueued_sequence_number,
+      partitionId: info.partition,
+      type: info.type
+    };
+    debug(`[%s] The partition info is: ${partitionInfo}.`, connection.options.id);
+    return partitionInfo;
+  }
+
+  /**
+   * Closes the AMQP management session to the Event Hub for this client,
+   * returning a promise that will be resolved when disconnection is completed.
+   * @return {Promise<void>}
+   */
+  async close(): Promise<void> {
+    try {
+      if (this._mgmtReqResLink) {
+        await rheaPromise.closeSession(this._mgmtReqResLink.session);
+        debug("Successfully closed the management session.");
+        this._mgmtReqResLink = undefined;
+      }
+    } catch (err) {
+      const msg = `An error occurred while closing the management session: ${err}`;
+      debug(msg);
+      throw new Error(msg);
+    }
+  }
+
+  private async _init(connection: any, endpoint: string, replyTo: string): Promise<void> {
+    if (!this._mgmtReqResLink) {
+      const rxopt: rheaPromise.ReceiverOptions = { source: { address: endpoint }, name: replyTo, target: { address: replyTo } };
+      debug("Creating a session for $management endpoint");
+      this._mgmtReqResLink = await createRequestResponseLink(connection, { target: { address: endpoint } }, rxopt);
+      debug(`[${connection.options.id}] Created sender "${this._mgmtReqResLink.sender.name}" and receiver "${this._mgmtReqResLink.receiver.name}" links for $management endpoint.`);
+    }
+  }
+
+  /**
+   * @private
+   * Helper method to make the management request
+   * @param {Connection} connection - The established amqp connection
+   * @param {string} type - The type of entity requested for. Valid values are "eventhub", "partition"
+   * @param {string | number} [partitionId] - The partitionId. Required only when type is "partition".
+   */
+  private async _makeManagementRequest(connection: any, type: "eventhub" | "partition", partitionId?: string | number): Promise<any> {
```

Comment:
>_makeManagementRequest [](start = 16, length = 22)

wouldn't this belong on the rpc class as a method? we have a state machien to manage this type of operations in the iothub sdk. i think it'd be worth it to explore in the future how to separate this thing into a component that both SDKs can use (maybe as one of the rhea extensions we've been talking about?)

Reply:
yes this is true. Will definitely do that.